proto/bucket/where_is.proto

```diff
@@ -54,6 +56,10 @@ message BucketItem {
 
   // Type of the resource
   ResourceIdentifier type = 4;
+
+  // References the schema and item within the bucket is associated with. Bucket items do not need to use the same schema reference
+  // if the bucket is "generic" // optional
+  sonrio.sonr.schema.SchemaReference schema_definition = 5;
```

Comment:
nit: name this `schema_reference` since "SchemaDefinition" is its own thing.

Reply:
refactored to just be the `schema_did`